Library/Homebrew/utils/pypi.rb

```diff
@@ -249,9 +249,16 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
       end
     end
 
+    python_deps = formula.deps
+                         .map(&:to_formula)
+                         .select { |f| f.name.start_with?("python@") }
+                         .sort_by(&:version)
+                         .reverse
+
     extra_packages = (extra_packages || []).map { |p| Package.new p }
     exclude_packages = (exclude_packages || []).map { |p| Package.new p }
-    exclude_packages += %w[argparse pip setuptools wsgiref].map { |p| Package.new p }
+    exclude_packages += %w[argparse pip wsgiref].map { |p| Package.new p }
+    exclude_packages.append(Package.new("setuptools")) if python_deps.first&.version&.< Version.new("3.12")
```

Comment:
How do we know `.first` is the right one here? Might be worth doing something like this too:
```suggestion
    exclude_packages.append(Package.new("setuptools")) if python_deps.first && python_deps.first.version < Version.new("3.12")
```

Reply:
`python_deps` is reverse sorted by `version`, so first should be the max which is what we want